### PR TITLE
Enhance ZHA device pairing feedback

### LINKED ISSUE #6620: Add "New device added" confirmation in ZHA
### The request
When pairing new devices to ZHA, there is no confirmation that the device is added after the card with the new device pops up. 
A suggestion would be to add confirmation, in green, at the top. The confirmation could say, ***"New device added!"*** and a checkmark next to it. The confirmation could stay there until the user leaves the window, or it could remain active for like 30 seconds. 

## Mockup image
ZHA new device confirmation

## Additional information

### PR DESCRIPTION
**Pull request by dmulcahey**
# Proposed change

This PR enhances the visual feedback presented to the user when pairing devices in the ZHA configuration panel. Feedback is given as soon as a device joins the network and pairing progress is updated as the device is interviewed and configured.

Fixes: #6620

**Requires:** https://github.com/home-assistant/core/pull/43729

device_pairing

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #6620
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by JuanMTech** [comment]
Definitely a must add

File: src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts [hunk 1]
```diff
@@ -175,8 +174,15 @@ class ZHAAddDevicesPage extends LitElement {
...
     }
-    if (message.type && message.type === "device_fully_initialized") {
-      this._discoveredDevices.push(message.device_info);
+    if (
+      message.type &&
+      [
+        "device_joined",
+        "raw_device_initialized",
+        "device_fully_initialized",
+      ].includes(message.type)
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts:183]: Can we create a Const above the class for these message types

Comment (zsarnett) [reply src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts:183]: Actually to the zha data file

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.device) {
+      return html``;
+    }
+
+    return html`
+      <ha-card
+        outlined
+        class=${this.device.pairing_status === "INITIALIZED"
```
Comment (zsarnett) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:42]: we should add all of these Strings as variables to the ZHA data file. Same for all of the strings below
```ts
const INITIALIZED = "INITIALIZED";
```

File: src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts (hunk 1 above)
Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts]: done

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts (hunk 2 above)
Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts]: done

File: src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ class ZHAAddDevicesPage extends LitElement {
 
   @internalProperty() private _error?: string;
 
-  @internalProperty() private _discoveredDevices: ZHADevice[] = [];
+  @internalProperty() private _discoveredDevices = {};
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts:37]: Please add typing

Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts:37]:
```suggestion
  @internalProperty() private _discoveredDevices: Record<string, ZHADevice> = {};
```

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+          </h1>
+          <h4>
+            ${this.hass!.localize(
+              `ui.panel.config.zha.device_pairing_card.${this.device.pairing_status}_status_text`
+            )}
+          </h4>
+        </div>
+        <div class="card-content">
+          ${this.device.pairing_status === INTERVIEW_COMPLETE ||
+          this.device.pairing_status === CONFIGURED
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:65]:
```suggestion
          ${[INTERVIEW_COMPLETE, CONFIGURED].inlcudes(this.device.pairing_status)
```

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    if (!this.hass || !this.device) {
+      return html``;
+    }
+
+    return html`
+      <ha-card
+        outlined
+        class=${this.device.pairing_status === INITIALIZED
+          ? "discovered"
+          : "discovered-progress"}
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:50]:
```suggestion
        class=${classMap({discovered: this.device.pairing_status === INITIALIZED})}
```

File: src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts (hunk 3 above)
Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts]: done

Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/zha-add-devices-page.ts]: done

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts (hunk 4 above)
Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts]: done

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts (hunk 5 above)
Comment (dmulcahey) [reply src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts]: I think I did this right... lmk if this needs more changes

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 6]
```diff
@@ -45,10 +46,14 @@ class ZHADevicePairingStatusCard extends LitElement {
...
-          ? "discovered"
-          : "discovered-progress"}
-        ><div class="header">
+        class="discovered-progress ${classMap({
+          discovered: this.device.pairing_status === INITIALIZED,
+        })}"
+        ><div
+          class="header ${classMap({
+            success: this.device.pairing_status === INITIALIZED,
+          })}"
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:55]: No need for this second class, it is already on the card.

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this.device) {
+      return html``;
+    }
+
+    return html`
+      <ha-card
+        outlined
+        class="discovered-progress ${classMap({
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:49]:
```suggestion
        class="discovered ${classMap({
```

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+          discovered: this.device.pairing_status === INITIALIZED,
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:50]:
```suggestion
          initialized: this.device.pairing_status === INITIALIZED,
```

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+        })}"
+        ><div
+          class="header ${classMap({
+            success: this.device.pairing_status === INITIALIZED,
+          })}"
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:55]:
```suggestion
          class="header"
```

File: src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+        .discovered .header {
+          background: var(--primary-color);
+          color: var(--text-primary-color);
+          padding: 8px;
+          text-align: center;
+          margin-bottom: 20px;
+        }
+        .discovered .header.success {
+          background: var(--success-color);
+        }
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/zha-device-pairing-status-card.ts:129]:
```suggestion
        .discovered {
          --ha-card-border-color: var(--primary-color);
        }
        .discovered.initialized {
          --ha-card-border-color: var(--success-color);
        }
        .discovered .header {
          background: var(--primary-color);
          color: var(--text-primary-color);
          padding: 8px;
          text-align: center;
          margin-bottom: 20px;
        }
        .discovered.initialized .header {
          background: var(--success-color);
        }
```

**Review by bramkragten — APPROVED** [review]

**Comment by dmulcahey** [comment]
backend is merged for this.

**Pull request merged**